### PR TITLE
change some code-blocks to runnableExamples and correct some errors in docs

### PR DESCRIPTION
**Pull request by ringabout**

### CODE REVIEW

File: lib/pure/json.nim [hunk 1]
```diff
@@ -519,9 +519,9 @@ proc `{}`*(node: JsonNode, keys: varargs[string]): JsonNode =
   ## This proc can be used to create tree structures on the
   ## fly (sometimes called `autovivification`:idx:):
   ##
-  ## .. code-block:: nim
-  ##   myjson{"parent", "child", "grandchild"} = newJInt(1)
-  ##
+  runnableExamples:
+    doAssert myjson{"parent", "child", "grandchild"} == newJInt(1)
```
Comment (ghost) [inline lib/pure/json.nim:524]: this one is obviously wrong since myjson isn't defined here

Comment (ringabout) [reply lib/pure/json.nim:524]: done

File: lib/pure/pathnorm.nim [hunk 2]
```diff
@@ -100,15 +100,12 @@ proc addNormalizePath*(x: string; result: var string; state: var int;
...
-  ## .. code-block:: nim
-  ##   assert normalizePath("./foo//bar/../baz") == "foo/baz"
-  ##
-  ##
   ## - Turns multiple slashes into single slashes.
   ## - Resolves '/foo/../bar' to '/bar'.
   ## - Removes './' from the path (but "foo/.." becomes ".")
+  ##
+  runnableExamples:
+    doAssert normalizePath("./foo//bar/../baz") == "foo/baz"
```
Comment (ringabout) [inline lib/pure/pathnorm.nim]: In windows `normalizePath("./foo//bar/../baz") == "foo\baz"`, the whole original comments are wrong.

Comment (ringabout) [reply lib/pure/pathnorm.nim]: I will use `when defined(posix)`.

File: lib/pure/pathnorm.nim [hunk 3]
```diff
@@ -100,15 +100,13 @@ proc addNormalizePath*(x: string; result: var string; state: var int;
...
-  ## - Turns multiple slashes into single slashes.
-  ## - Resolves '/foo/../bar' to '/bar'.
-  ## - Removes './' from the path (but "foo/.." becomes ".")
+  runnableExamples:
+    when defined(posix):
+      # - Turns multiple slashes into single slashes.
+      # - Resolves '/foo/../bar' to '/bar'.
+      # - Removes './' from the path (but "foo/.." becomes ".")
+      doAssert normalizePath("./foo//bar/../baz") == "foo/baz"
+
```
Comment (juancarlospaco) [inline lib/pure/pathnorm.nim]:
```suggestion
      doAssert normalizePath("./foo//bar/../baz") == "foo/baz"

  ## - Turns multiple slashes into single slashes.
  ## - Resolves `'/foo/../bar'` to `'/bar'`.
  ## - Removes `'./'` from the path, but `"foo/.."` becomes `"."`.
```

Theres no reason for the comments to be inside the code, and the code examples should be as minimal as possible I think.
Looks good, nice work.

**Comment by ringabout** [comment]
CI failure
```
2020-11-11T01:45:46.6305901Z FAIL: tests/misc/tsizeof.nim C++
2020-11-11T01:45:46.6483176Z Test "tests\misc\tsizeof.nim" in category "misc"
2020-11-11T01:45:46.7047724Z Failure: reNimcCrash
2020-11-11T01:45:46.7392487Z $ D:\a\1\s\bin\nim.exe cpp --hints:on -d:testing --nimblePath:tests/deps --nimCache:nimcache\tests\misc\tsizeof.nim_f6f87c9fdcf8b3c3f07f93f1ee8712c9 tests\misc\tsizeof.nim
2020-11-11T01:45:46.7539478Z Hint: used config file 'D:\a\1\s\config\nim.cfg' [Conf]
2020-11-11T01:45:46.7708848Z Hint: used config file 'D:\a\1\s\config\config.nims' [Conf]
2020-11-11T01:45:46.8392155Z Hint: used config file 'D:\a\1\s\tests\config.nims' [Conf]
2020-11-11T01:45:46.9631927Z ......
2020-11-11T01:45:46.9982217Z tsizeof.nim(168, 7) Hint: 'debug' is declared but not used [XDeclaredButNotUsed]
2020-11-11T01:45:47.0402515Z tsizeof.nim(374, 9) Hint: 'trivialSize' is declared but not used [XDeclaredButNotUsed]
2020-11-11T01:45:47.0601047Z tsizeof.nim(374, 9) Hint: 'trivialSize' is declared but not used [XDeclaredButNotUsed]
2020-11-11T01:45:47.1201598Z CC: stdlib_assertions.nim
2020-11-11T01:45:47.1893066Z CC: stdlib_io.nim
2020-11-11T01:45:47.2846409Z CC: stdlib_system.nim
2020-11-11T01:45:47.3894244Z CC: tsizeof.nim
2020-11-11T01:45:47.4791280Z Hint:  [Link]
2020-11-11T01:45:47.5873606Z H
```

**Pull request merged**